tools/continuous-ingestor/timestream_sample_continuous_data_ingestor_application.py

```diff
@@ -131,25 +135,36 @@
 measureValuesForTaskEndState = ['SUCCESS_WITH_NO_RESULT', 'SUCCESS_WITH_RESULT', 'INTERNAL_ERROR', 'USER_ERROR', 'UNKNOWN', 'THROTTLED']
 selectionProbabilities = [0.2, 0.7, 0.01, 0.07, 0.01, 0.01]
 
-DimensionsMetric = namedtuple('DimensionsMetric', 'region cell silo availability_zone microservice_name instance_type os_version instance_name')
-DimensionsEvent = namedtuple('DimensionsEvent', 'region cell silo availability_zone microservice_name instance_name process_name, jdk_version')
+DimensionsMetric = namedtuple('DimensionsMetric', 'region cell silo availability_zone microservice_name instance_name instance_type os_version')
+DimensionsEvent = namedtuple('DimensionsEvent', 'region cell silo availability_zone microservice_name instance_name process_name jdk_version')
 
 def generateRandomAlphaNumericString(length = 5):
     rand = random.Random(12345)
     x = ''.join(rand.choice(string.ascii_letters + string.digits) for x in range(length))
     return x
 
-def generateDimensions(scaleFactor):
+def generateDimensions(regionINC, microserviceINC, scaleFactor):
     instancePrefix = generateRandomAlphaNumericString(8)
     dimensionsMetrics = list()
     dimenstionsEvents = list()
-
-    for region in regions:
+    
+    myRegions = regions.copy()
+    if ( len(regionINC) > 0):
```

Comment:
The Pythonic expression here is simply `if regionINC:`, which checks both that the collection is `None` and non-empty. You actually don't need the `copy` or the `list` here, so you can do
```
myRegions = set(regions) & set(regionINC) if regionINC else regions
```

ditto for `myMicroservices` below